pkg/installmanager/installmanager.go

```diff
@@ -1803,10 +1807,119 @@ func (m *InstallManager) overrideCredentialsModeOnInstallConfig(credentialsMode
+	if modifiedBytes != nil {
+		err = ioutil.WriteFile(overrideCredsPath, *modifiedBytes, 0644)
+		if err != nil {
+			return errors.Wrap(err, "error writing 99_cloud-creds-secret.yaml")
+		}
+		m.log.Info("patched 99_cloud-creds-secret.yaml successfully")
+	}
+
 	return nil
 }
+
+func patchAzureOverrideCreds(overrideCredsBytes, clusterInfraConfigBytes []byte, region string) (*[]byte, error) {
+	overrideCredsJson, err := yaml.YAMLToJSON(overrideCredsBytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "error converting 99_cloud-creds-secret.yaml to json")
+	}
+
+	regionInput := gjson.Get(string(overrideCredsJson), `data.azure_region`).String()
+	infraNameInput := gjson.Get(string(overrideCredsJson), `data.azure_resource_prefix`).String()
+	resourceGroupNameInput := gjson.Get(string(overrideCredsJson), `data.azure_resourcegroup`).String()
+
+	//Return error if any of infraName/resourceGroupName already exist
+	if infraNameInput != "" || resourceGroupNameInput != "" {
+		return nil, fmt.Errorf("azure_resource_prefix/azure_resourcegroup already exists in %s", overrideCredsYAML)
+	}
+
+	//Return error if region exists but different to the one in cd.spec
+	if regionInput != "" {
+		regionDecode, err := base64.StdEncoding.DecodeString(regionInput)
+		if err != nil {
+			return nil, errors.Wrap(err, "error decoding data.azure_region")
+		}
+		if region != string(regionDecode) {
+			return nil, fmt.Errorf("azure_region=%s already exists in 99_cloud-creds-secret.yaml but different to region in ClusterDeployment.spec", string(regionDecode))
+		}
+	}
+
+	clusterInfraConfigJson, err := yaml.YAMLToJSON(clusterInfraConfigBytes)
+	if err != nil {
+		return nil, errors.Wrap(err, "error converting cluster-infrastructure-02-config.yml to json")
+	}
+
+	infraName := gjson.Get(string(clusterInfraConfigJson), `status.infrastructureName`).String()
+	resourceGroupName := gjson.Get(string(clusterInfraConfigJson), `status.platformStatus.azure.resourceGroupName`).String()
+	if region == "" || infraName == "" || resourceGroupName == "" {
+		return nil, fmt.Errorf("error reading from manifests, region=%s, clusterInfraConfig=%s", region, string(clusterInfraConfigJson))
+	}
+
+	regionBase64 := interface{}(base64.StdEncoding.EncodeToString([]byte(region)))
+	infraNameBase64 := interface{}(base64.StdEncoding.EncodeToString([]byte(infraName)))
+	resourceGroupNameBase64 := interface{}(base64.StdEncoding.EncodeToString([]byte(resourceGroupName)))
+
+	ops := yamlpatch.Patch{
+		yamlpatch.Operation{
+			Op:    "add",
+			Path:  yamlpatch.OpPath("/data/azure_region"),
+			Value: yamlpatch.NewNode(&regionBase64),
+		},
+		yamlpatch.Operation{
+			Op:    "add",
+			Path:  yamlpatch.OpPath("/data/azure_resource_prefix"),
+			Value: yamlpatch.NewNode(&infraNameBase64),
+		},
+		yamlpatch.Operation{
+			Op:    "add",
+			Path:  yamlpatch.OpPath("/data/azure_resourcegroup"),
+			Value: yamlpatch.NewNode(&resourceGroupNameBase64),
+		},
+	}
+	if regionInput != "" {
+		ops = ops[1:3]
+	}
```

Comment:
ah, this would mitigate the above.

IMO the code would be a bit prettier if the logic were inverted.
```go
ops := yamlpatch.Patch{ ... resource_prefix and resourcegroup operations ...}
if regionInput == "" {
    ops = append(ops, ... region operation ...)
}
```